pkg/app/launcher/cmd/launcher/launcher.go

```diff
@@ -267,12 +267,17 @@ func (l *launcher) run(ctx context.Context, input cli.Input) error {
 
 	group.Go(func() error {
 		// Execute the first time immediately.
-		execute()
+		if err := execute(); err != nil {
+			input.Logger.Error("LAUNCHER: failed while launching new Piped", zap.Error(err))
+			// Return an error if the initial startup fails.
+			return err
+		}
 
 		for {
 			select {
 			case <-ticker.C:
-				execute()
+				// Don't return an error to continue piped execution.
+				_ = execute()
```

Comment:
```suggestion
				execute()
```
nits
looks like this is caused by some lint, lets remove it to unify with others 👍